code/modules/projectiles/guns/pistols.dm

```diff
@@ -119,6 +119,17 @@
 	to_chat(attacher, span_warning("You cannot attach [src] to [attaching_to] while [attachments_by_slot[ATTACHMENT_SLOT_RAIL]] occupies [src]'s rail slot."))
 	return FALSE
 
+/obj/item/weapon/gun/pistol/plasma_pistol/on_attach(obj/item/attached_to, mob/user)
+	if(!(flags_gun_features & GUN_WIELDED_FIRING_ONLY))
+		flags_gun_features += GUN_WIELDED_FIRING_ONLY
+	. = ..()
+
+/obj/item/weapon/gun/pistol/plasma_pistol/on_detach(obj/item/attached_to, mob/user)
+	if(flags_gun_features & GUN_WIELDED_FIRING_ONLY )
+		flags_gun_features -= GUN_WIELDED_FIRING_ONLY
+	. = ..()
```

Comment:
use return ..() instead of . = ..() at the end of procs and use the proper bitfield toggles